Mu2eG4/src/constructHall.cc

```diff
@@ -122,25 +125,27 @@ namespace mu2e {
 
       geoOptions->loadEntry( config, volume.getName(), volume.getName() );
 
-      if ( notchMgr.hasNotches( volName ) ) {
-        // First do volumes with notches
+      if ( notchMgr.hasNotches( volName )||notchMgr.hasHoles( volName ) ) {
+       vector<Notch> volNotches = notchMgr.getNotchVector(volName);
```

Comment:
I know that you inheritted this.  On these two lines, please change the RHS variables to const&.  As written it makes an unnecessary deep copy.

Reply:
const vector<Notch>& volNotches = notchMgr.getNotchVector(volName);
const vector<Hole>& volHoles = notchMgr.getHoleVector(volName);
I hope these are the changes required to avoid deep copy. I made the same changes in constructTrapSolids loop as well.

Reply:
Yes, that's what I wanted.  Thanks.